src/otx/algo/classification/torchvision_model.py

```diff
@@ -451,219 +164,130 @@ def __init__(
-            "hierarchical": self.task == OTXTaskType.H_LABEL_CLS,
-        }
-
-        return super()._export_parameters.wrap(**export_params)
-
-    @property
-    def _exporter(self) -> OTXModelExporter:
-        """Creates OTXModelExporter object that can export the model."""
-        return OTXNativeModelExporter(
-            task_level_export_parameters=self._export_parameters,
-            input_size=(1, 3, *self.input_size),
-            mean=(123.675, 116.28, 103.53),
-            std=(58.395, 57.12, 57.375),
-            resize_mode="standard",
-            pad_value=0,
-            swap_rgb=False,
-            via_onnx=False,
-            onnx_export_configuration=None,
-            output_names=["logits", "feature_vector", "saliency_map"] if self.explain_mode else None,
+
+    def forward_explain(self, inputs: MultilabelClsBatchDataEntity) -> MultilabelClsBatchPredEntity:
+        """Model forward explain function."""
+        outputs = self.model(images=inputs.stacked_images, mode="explain")
+
+        return MultilabelClsBatchPredEntity(
+            batch_size=len(outputs["preds"]),
+            images=inputs.images,
+            imgs_info=inputs.imgs_info,
+            labels=outputs["preds"],
+            scores=outputs["scores"],
+            saliency_map=outputs["saliency_map"],
+            feature_vector=outputs["feature_vector"],
         )
 
-    def training_step(self, batch: CLASSIFICATION_BATCH_DATA_ENTITY, batch_idx: int) -> Tensor:
-        """Performs a single training step on a batch of data.
+    def forward_for_tracing(self, image: torch.Tensor) -> torch.Tensor | dict[str, torch.Tensor]:
+        """Model forward function used for the model tracing during model exportation."""
+        if self.explain_mode:
+            return self.model(images=image, mode="explain")
 
-        Args:
-            batch (MulticlassClsBatchDataEntity): The input batch of data.
-            batch_idx (int): The index of the current batch.
+        return self.model(images=image, mode="tensor")
 
-        Returns:
-            Tensor: The computed loss for the training step.
-        """
-        loss = super().training_step(batch, batch_idx)
-        # Collect metrics related to Semi-SL Training.
-        if self.train_type == OTXTrainType.SEMI_SUPERVISED:
-            self.log(
-                "train/unlabeled_coef",
-                self.model.head.unlabeled_coef,
-                on_step=True,
-                on_epoch=False,
-                prog_bar=True,
-            )
-            self.log(
-                "train/num_pseudo_label",
-                self.model.head.num_pseudo_label,
-                on_step=True,
-                on_epoch=False,
-                prog_bar=True,
-            )
-        return loss
 
-    def forward_explain(self, inputs: CLASSIFICATION_BATCH_DATA_ENTITY) -> CLASSIFICATION_BATCH_PRED_ENTITY:
+class TVModelForHLabelCls(OTXHlabelClsModel):
```

Comment:
Could you update arguemnt docstring?

Reply:
Thanks! Done - https://github.com/openvinotoolkit/training_extensions/commit/5ed3863c2f7d3db9b94acbbf9530abbce28536ba